### DISCUSSION TITLE
`Zfinx` support in GCC `v13.2.0`

### DISCUSSION THREAD

**Discussion by NikLeberg** [stnolting/neorv32#765] [Ideas]
I just stumbled over the fact that GCC in version `13.2.0` [seems to support](https://gcc.gnu.org/gcc-13/changes.html#:~:text=Zdinx-,Zfinx,-Zfh) the `Zfinx` extension from C code! 🎇
See: https://godbolt.org/z/61hx5hGs7

... so me might want to document that 😊.
What do you think @stnolting ?

**Comment by mikaelsky** [comment]
We've been building with zfinx support for a while and beyond adding "zfinx" to the march you also need to extend CC_OPTS with -mno-fdiv to prevent GCC from calling the zfinx hardware for DIV and SQRT instructions and instead inject the soft_float library version of the same commands.

**Comment by stnolting** [comment]
> ... so me might want to document that 😊.

Absolutely! But before doing that I would like to "complete" the FPU since there are still some TODOs:
* Subnormal numbers are not supported (they're flushed to zero). For many applications this might be fine and this also does not interfere with the ISA specs. But soon there will be `Zfinx`-support in [riscv-arch-test][https://github.com/riscv-non-isa/riscv-arch-test] and I would like to use that to further test the FPU - and there might be some trouble with those subnormal numbers.
* Division and square root operations are not implemented yet. But as @mikaelsky has mentioned we can circumvent that (for now).
* The exception flags are untested and I think there are still some bugs in there.

**Reply by mikaelsky** [reply]
We do have a... non-official version of the Zfinx test vectors based on the riscv-arch-test F test which we have been hammering the cores FPU with a lot of things are indeed working.. with the exception of the exceptions (pun)

The test can be shared, its basically some script modifications to the riscv-arch-test for the F tests followed by some post processing to deal with overlapping register assignments. The easiest might be to just share the post-processed .S files and the output signature for comparison? I believe you are doing a post sim comparison and not an in testbench comparison as we are doing on our end?

It still on my list of tasks to walk through all the Zfinx test to find+fix any exception flag issues.
